fxpubsub/module_test.go

```diff
@@ -0,0 +1,145 @@
+package fxpubsub_test
+
+import (
+	"context"
+	"testing"
+
+	fxpubsub "github.com/ankorstore/yokai-contrib/module/pubsub"
+	"github.com/ankorstore/yokai/config"
+	"github.com/ankorstore/yokai/fxconfig"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/fx"
+	"go.uber.org/fx/fxtest"
+)
+
+// TestFxPubSubModule verifies that the FxPubSubModule can be loaded without errors.
+func TestFxPubSubModule(t *testing.T) {
+	t.Setenv("APP_CONFIG_PATH", "testdata/config")
+
+	var conf *config.Config
+
+	app := fxtest.New(
+		t,
+		fx.NopLogger,
+		fxpubsub.FxPubSubModule,
+		fxconfig.FxConfigModule,
+		fx.Populate(&conf),
+	)
+
+	// Start and stop the application, checking for errors.
+	err := app.Start(context.Background())
+	assert.NoError(t, err, "failed to start the Fx application")
+
+	err = app.Stop(context.Background())
+	assert.NoError(t, err, "failed to stop the Fx application")
+}
+
+// TestNewFxPubSub verifies that NewFxPubSub returns a valid pubsub.Client for test without ProjectID.
+func TestNewFxPubSubForTestClientWithoutProjectID(t *testing.T) {
+	t.Setenv("APP_CONFIG_PATH", "testdata/config")
+
+	var conf *config.Config
+
+	app := fxtest.New(
+		t,
+		fx.NopLogger,
+		fxpubsub.FxPubSubModule,
+		fxconfig.FxConfigModule,
+		fx.Populate(&conf),
+	).RequireStart()
+
+	param := fxpubsub.FxPubSubParam{
+		Config: conf,
+	}
+
+	client, err := fxpubsub.NewFxPubSub(param)
```

Comment:
`NewFxPubSub` is called by loading the module, no need to call it externally

we better get fxtest start error and assert we have the error related to the missing project id